ClassicPress/ClassicPress-v2
# Add fallback Update URI when parsing plugin data

## Description
Add a fallback for empty `Update URI` to ClassicPress directory URL if `Requires CP` is set.

Original PR on 1.x: https://github.com/ClassicPress/ClassicPress/pull/1221

## Motivation and context
As discussed [in the forum](https://forums.classicpress.net/t/plugin-theme-updates-directory-integration/4558) if `Requires CP` is set but `Update URI` is empty, fall back to the ClassicPress directory URL with the slug set to the plugin's base directory name.

## How has this been tested?
**Added Unit tests for this change.**
To test manually:
- Install and activate the [directory integration plugin](https://github.com/ClassicPress/classicpress-directory-integration)
- Change https://directory.classicpress.net to https://staging-directory.classicpress.net in this PR
- Pick a plugin from https://staging-directory.classicpress.net
- Check that it have the `Requires CP` header field and don't have `Update URI` (swap those conditions to test)
- Edit the plugin version to a lower number

In plugins page an update shows up.

## Types of changes
- New feature
